### PR TITLE
[MRG] refactor and add tests for containment direction for ANI calculation

### LINKED ISSUE #1835: How to read the containment matrix?
Hi,

If I got a containment matrix as:
     a     b
a   1     0.3
b   0.6  1

Am I right that C(a,b) = 0.3, and C(b,a) = 0.6? Or instead the reverse is true?

Thanks!

Zewei

### LINKED ISSUE #2193: check that reported `search` containment ani is using the right containment direction
for @ctb -

If we're running `sourmash search`, are we producing the containment of the query in the reference, or the reference in the query? Thinking about it now I think we're doing query in the reference.

Over here https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/search.py#L324, I think we're getting containment ANI relative to query (but using containment we generated during search), which would be wrong.

Generally, I think this is related to the syntax/meanings of `containment_ani` vs `contained_by`.

`self.contained_by(other)` = containment of self in other
`self.containment_ani` = ANI from containment of other in self

I think, besides fixing, it would probably be helpful to standardize the python api -- maybe `contained_by_to_ani`?

I think `prefetch` and `gather` results should be fine, because we generate containment, ani within those, so it was clear which direction to go.

### LINKED ISSUE #2194: document containment direction for `compare` (and standardize for containment ani)
When running `sourmash compare`, the matrix will be asymmetric. But which direction is which? And is it the same for `containment` and `containment-ani`?

Based on code here: https://github.com/sourmash-bio/sourmash/blob/latest/src/sourmash/compare.py#L75-L90

For `--containment`, we do `containments[i][j] = siglist[j].contained_by(siglist[i])`
while for `--containment-ani`, we do `siglist[j].containment_ani(siglist[i])`

...which will give accurate values for each but populate the matrix in _opposite_ directions (ref #2193) This directional change will likely be fixed by changes for #2193, but we should document the final shared direction.

Sticking with the `--containment` direction, with `i`=row, `j`=col, the matrix will be the column genome contained by the row genome.

### PR DESCRIPTION
**Pull request by ctb**
This PR confirms and adds tests for the direction of containment for the ANI calculation, per #2193, and documents the `compare` matrix output order for asymmetric containment comparisons. It also changes the ANI code to use the following naming scheme:

* `mh1_containment_ani` became `ani_from_mh1_containment_in_mh2`
* `estimate_mh1_containment_ani` became `estimate_ani_from_mh1_containment_in_mh2`

and so on, as discussed in https://github.com/sourmash-bio/sourmash/issues/2193.

Fixes https://github.com/sourmash-bio/sourmash/issues/2193
Fixes https://github.com/sourmash-bio/sourmash/issues/2194
Fixes https://github.com/sourmash-bio/sourmash/issues/1835

TODO:
- [x] rename `containment_ani` to something clearer
- [x] write more tests w/ bigger asymmetries for search output
- [x] write `compare --ani` test
- [x] (add asymmetric test from #1835 maybe?)
- [x] update `compare` documentation

### CODE REVIEW

**Comment by ctb** [comment]
so @bluegenes I was struck by the fact that only one test, `tests/test_sourmash.py::test_search_ani_containment_estimate_ci`, was broken by this fix, and that the actual ANI estimate didn't change, just the CI. It's fantastic that a test broke, but I'd like a clearer example of where this matters ;).

I was thinking that I could write a few more tests - specifically,
* one where the containment ANI is quite asymmetric
* one for the compare matrix to verify the values

If this seems like a bad idea, speak ~now-ish or forever hold your peace :)

**Comment by ctb** [comment]
ready for review & merge @bluegenes!

File: doc/command-line.md [hunk 1]
```diff
@@ -209,12 +213,27 @@ Options:
...
+The containment matrix is organized such that the value in row A for column B is the containment of the B'th sketch in the A'th sketch, i.e.
+
+```
+C(A, B) = B.contained_by(A)
+```
+
+**Note:** The ANI estimate will be calculated based on Jaccard similarity
+by default; however, if `--containment` or `--max-containment` is
+specified, those values will be used instead. With `--containment --ani`, the
+ANI output matrix will be asymmetric as discussed above.
```
Comment (bluegenes) [inline doc/command-line.md]:
```suggestion
by default; however, if `--containment`, `--max-containment`, or `--avg-containment` is
specified, those values will be used instead. With `--containment --ani`, the
ANI output matrix will be asymmetric as discussed above.
```

File: tests/test_sourmash.py [hunk 2]
```diff
@@ -5802,8 +5803,28 @@ def test_search_ani_containment(c):
...
+
+    with sourmash_args.FileInputCSV(runtmp.output('query-in-merged.csv')) as r:
+        query_in_merged = list(r)[0]
+
+    with sourmash_args.FileInputCSV(runtmp.output('merged-in-query.csv')) as r:
+        merged_in_query = list(r)[0]
+
+    assert query_in_merged['ani'] == '1.0'
+    assert merged_in_query['ani'] == '0.9865155060423993'
+
```
Comment (bluegenes) [inline tests/test_sourmash.py:5824]: this is the one I thought would be wrong ... I'm convinced.

**Review by bluegenes — APPROVED** [review]
Thank you for going through and checking everything, as well as writing tests with increased asymmetry. Not a huge fan of the extremely verbose names, but at least they're clear!

File: tests/test_sourmash.py (hunk 2 above)
Comment (ctb) [reply tests/test_sourmash.py:5824]: 🎉

**Pull request merged**